### ISSUE TITLE
Error on colab notebook

### ISSUE THREAD

**Issue by bamps53** [EmGarr/kerod#1]
**Describe the bug**
Colab notebook stop by follwoing error.

**To Reproduce**
Just run all this notebook.
https://colab.research.google.com/github/Emgarr/kerod/blob/master/notebooks/smca_coco_training_multi_gpu.ipynb

**Expected behavior**

**Screenshots**
```
INFO:tensorflow:Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
INFO:tensorflow:Using MirroredStrategy with devices ('/job:localhost/replica:0/task:0/device:GPU:0',)
Epoch 1/50
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
WARNING:tensorflow:Using a while_loop for converting EagerPyFunc
INFO:tensorflow:Error reported to Coordinator: minimize() got an unexpected keyword argument 'tape'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/coordinator.py", line 297, in stop_on_exception
    yield
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_run.py", line 323, in run
    self.main_result = self.main_fn(*self.main_args, **self.main_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 667, in wrapper
    return converted_call(f, args, kwargs, options=options)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 396, in converted_call
    return _call_unconverted(f, args, kwargs, options)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 478, in _call_unconverted
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 788, in run_step
    outputs = model.train_step(data)
  File "/usr/local/lib/python3.6/dist-packages/kerod/model/smca_detr.py", line 321, in train_step
    self.optimizer.minimize(loss, self.trainable_variables, tape=tape)
TypeError: minimize() got an unexpected keyword argument 'tape'
INFO:tensorflow:Error reported to Coordinator: minimize() got an unexpected keyword argument 'tape'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/training/coordinator.py", line 297, in stop_on_exception
    yield
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_run.py", line 323, in run
    self.main_result = self.main_fn(*self.main_args, **self.main_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 667, in wrapper
    return converted_call(f, args, kwargs, options=options)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 396, in converted_call
    return _call_unconverted(f, args, kwargs, options)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/autograph/impl/api.py", line 478, in _call_unconverted
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 788, in run_step
    outputs = model.train_step(data)
  File "/usr/local/lib/python3.6/dist-packages/kerod/model/smca_detr.py", line 321, in train_step
    self.optimizer.minimize(loss, self.trainable_variables, tape=tape)
TypeError: minimize() got an unexpected keyword argument 'tape'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-3de9ded0293d> in <module>()
     28 ]
     29 
---> 30 history = model.fit(ds_train, validation_data=ds_val, epochs=50, callbacks=callbacks)
     31 model.save('detr_an_awesome_model')
     32 model.save_weights('detr_an_awesome_model.h5')

9 frames
/usr/local/lib/python3.6/dist-packages/tensorflow/python/framework/func_graph.py in wrapper(*args, **kwargs)
    975           except Exception as e:  # pylint:disable=broad-except
    976             if hasattr(e, "ag_error_metadata"):
--> 977               raise e.ag_error_metadata.to_exception(e)
    978             else:
    979               raise

TypeError: in user code:

    /usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py:805 train_function  *
        return step_function(self, iterator)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py:795 step_function  **
        outputs = model.distribute_strategy.run(run_step, args=(data,))
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/distribute_lib.py:1259 run
        return self._extended.call_for_each_replica(fn, args=args, kwargs=kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/distribute_lib.py:2730 call_for_each_replica
        return self._call_for_each_replica(fn, args, kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_strategy.py:629 _call_for_each_replica
        self._container_strategy(), fn, args, kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_run.py:93 call_for_each_replica
        return _call_for_each_replica(strategy, fn, args, kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_run.py:234 _call_for_each_replica
        coord.join(threads)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/coordinator.py:389 join
        six.reraise(*self._exc_info_to_raise)
    /usr/local/lib/python3.6/dist-packages/six.py:703 reraise
        raise value
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/training/coordinator.py:297 stop_on_exception
        yield
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/distribute/mirrored_run.py:323 run
        self.main_result = self.main_fn(*self.main_args, **self.main_kwargs)
    /usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py:788 run_step  **
        outputs = model.train_step(data)
    /usr/local/lib/python3.6/dist-packages/kerod/model/smca_detr.py:321 train_step
        self.optimizer.minimize(loss, self.trainable_variables, tape=tape)

    TypeError: minimize() got an unexpected keyword argument 'tape'
```
**Desktop (please complete the following information):**
colab notebook

**Additional context**
Thanks for sharing great work. I was surprised at finding SMCA inplementation!

**Comment by EmGarr** [comment]
@bamps53 What version of tensorflow-addons do you have ? 
`tape`has been introduced in tensorflow 2.4 so you should use: tensorflow-addons-0.12.1 or the nightly

**Comment by bamps53** [comment]
Thanks for the quick answer!
It seems colab tfa version is still 0.8.3.
If I add `!pip install -U tensorflow_addons` at the top, it works for the error.

![image](https://user-images.githubusercontent.com/37071446/106767235-fe666280-667d-11eb-9756-c0dffb6cb9ea.png)

But then I got weird warning which says there is no GPU and after that session was crashed by run out of RAM. 
![image](https://user-images.githubusercontent.com/37071446/106769237-1212c880-6680-11eb-8937-a27a0e7c760f.png)
![image](https://user-images.githubusercontent.com/37071446/106769457-569e6400-6680-11eb-933a-02bbd61afe62.png)

I solved this by changing num_gpu=1 and eliminate mirrored_strategy, just FYI.
